### ISSUE TITLE
Add RFCS for datafusion

### ISSUE THREAD

**Issue by xudong963** [apache/arrow-datafusion#1397]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I think it's very good to have RFCs for datafusion. 

**Describe the solution you'd like**
Before developing a big and meaningful feature, it's better to have an RFC ticket. Maybe we used Google Docs to do the similar thing before, but if have an RFC ticket, we can discuss it easier and more conveniently, finally, we can merge the RFC to our codebase. These deposits will be a valuable part of datafusion. New developers will be easier to get to know datafusion.

**Describe alternatives you've considered**
RFC is also useful for developers to organize their thoughts, and reviewer reviews of code with an RFC will be easier

**Additional context**
such as https://github.com/cockroachdb/cockroach/tree/master/docs/RFCS
Later, I'll write an RFC about the subquery feature.

**Comment by houqp** [comment]
Thanks @xudong963 for bringing this up, we had some prior discussions around RFCs in https://github.com/apache/arrow-datafusion/pull/422. The idea at the time was that we would use a less formal spec format to keep track of design proposals. All specs are published to the datafusion official website as well for more visibility.

**Comment by alamb** [comment]
In my opinion, getting major RFCs into a single place and the community used to finding and contributing to them would be a major win. 

My feeling / opinion is that at this point in DataFusion's process, the specific process for RFCs (e.g. the format, detail level, etc) can be quite loose. Getting the habit and communication started is the most important part -- we can hone the details as we go.

So, in other words, maybe we could add a note to the https://github.com/apache/arrow-datafusion/blob/ab41d8187d25b24af73d56d03ac166a5498922bc/docs/source/community/communication.md#contributing section (what is rendered at https://arrow.apache.org/datafusion/community/communication.html) pointing people at the RFCs?

**Comment by xudong963** [comment]
> Getting the habit and communication started is the most important part -- we can hone the details as we go.
Strongly agree, hone the details as we go! 



> So, in other words, maybe we could add a note to the https://github.com/apache/arrow-datafusion/blob/ab41d8187d25b24af73d56d03ac166a5498922bc/docs/source/community/communication.md#contributing section (what is rendered at https://arrow.apache.org/datafusion/community/communication.html) pointing people at the RFCs?

Yes, we can make a directory named doc/RFCs, and add signage in  https://github.com/apache/arrow-datafusion/blob/ab41d8187d25b24af73d56d03ac166a5498922bc/docs/source/community/communication.md#contributing to guide people to see our RFCs.
